src/pages/TermsOfService.ts

```diff
@@ -0,0 +1,149 @@
+import _ from 'lodash/fp';
+import { useState } from 'react';
+import { div, h, h1, img } from 'react-hyperscript-helpers';
+import { ButtonOutline, ButtonPrimary, ButtonSecondary } from 'src/components/common';
+import { centeredSpinner } from 'src/components/icons';
+import { MarkdownViewer, newWindowLinkRenderer } from 'src/components/markdown';
+import scienceBackground from 'src/images/science-background.jpg';
+import { Ajax } from 'src/libs/ajax';
+import { signOut } from 'src/libs/auth';
+import colors from 'src/libs/colors';
+import { reportError, withErrorReporting } from 'src/libs/error';
+import * as Nav from 'src/libs/nav';
+import { useOnMount, useStore } from 'src/libs/react-utils';
+import { authStore } from 'src/libs/state';
+import * as Style from 'src/libs/style';
+import * as Utils from 'src/libs/utils';
+
+const TermsOfServicePage = () => {
+  const [busy, setBusy] = useState<boolean>();
+  const { signInStatus, termsOfService } = useStore(authStore);
+  const acceptedLatestTos = signInStatus === 'signedIn' && termsOfService.userHasAcceptedLatestTos;
+  const usageAllowed = signInStatus === 'signedIn' && termsOfService.permitsSystemUsage;
+  const [tosText, setTosText] = useState<string>();
+
+  useOnMount(() => {
+    const loadTosAndUpdateState = _.flow(
+      Utils.withBusyState(setBusy),
+      withErrorReporting('There was an error retrieving our terms of service.')
+    )(async () => {
+      setTosText(await Ajax().User.getTos());
+    });
+    loadTosAndUpdateState();
+  });
+
+  const accept = async () => {
+    try {
+      setBusy(true);
+      const { enabled } = await Ajax().User.acceptTos();
+      if (enabled) {
+        const termsOfService = await Ajax().User.getTermsOfServiceComplianceStatus();
```

Comment:
The `UserStatus` object sent back from sam does not appear to be of this form. The `enabled` part of the JSON response is an object which corresponds to this Map:

`              val enabledMap = Map(
                "ldap" -> adminEnabled,
                "allUsersGroup" -> allUsersStatus,
                "google" -> googleStatus,
                "adminEnabled" -> adminEnabled,
                "tosAccepted" -> tosComplianceStatus.permitsSystemUsage
              )
`

So if the user is actually disabled, I believe this will still evaluate as true.